File: lib/ts/src/internal/validation/ValidateBoolean.ts
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

import { ValidationFailure } from '../../internal/validation/ValidationFailure.js';
import { getTypeUnexpectedValidationFailure } from '../../internal/validation/GetTypeUnexpectedValidationFailure.js';
import { booleanName } from '../types/TBoolean.js';

export function validateBoolean(value: any): ValidationFailure[] {
    if (typeof value === 'boolean') {
        return [];
    } else {
        return getTypeUnexpectedValidationFailure([], value, booleanName);
    }
}

File: lib/py/telepact/internal/validation/ValidateBoolean.py
#|
#|  Copyright The Telepact Authors
#|  SPDX-License-Identifier: Apache-2.0
#|

from typing import TYPE_CHECKING
from ..types.TBoolean import _BOOLEAN_NAME
from ...internal.validation.GetTypeUnexpectedValidationFailure import get_type_unexpected_validation_failure

if TYPE_CHECKING:
    from ...internal.validation.ValidationFailure import ValidationFailure


def validate_boolean(value: object) -> list['ValidationFailure']:
    if type(value) is bool:
        return []
    if isinstance(value, bool):
        return []
    return get_type_unexpected_validation_failure([], value, _BOOLEAN_NAME)
